scripts/DllManipulator.cs

```diff
@@ -48,20 +60,18 @@ internal static void Initialize(int unityMainThreadId, string assetsPath)
 
             LowLevelPluginManager.ResetStubPlugin();
 
-            Assembly[] assemblies;
-            if (Options.assemblyPaths.Length == 0)
-            {
-                assemblies = new[] { Assembly.GetExecutingAssembly() };
-            }
-            else
+            IEnumerable<string> assemblyPathsTemp = Options.assemblyNames;
+            if (!assemblyPathsTemp.Any())
+                assemblyPathsTemp = DEFAULT_ASSEMBLY_NAMES;
+            
+            assemblyPathsTemp = assemblyPathsTemp.Concat(INTERNAL_ASSEMBLY_NAMES);
+            
+            var allAssemblies = AppDomain.CurrentDomain.GetAssemblies();
+            var assemblies = allAssemblies.Where(a => !a.IsDynamic && assemblyPathsTemp.Any(p => p == Path.ChangeExtension(a.ManifestModule.Name, null))).ToArray();
```

Comment:
Is `ManifestModule.Name` guaranteed to be file name without path? If not, wouldn't it be better to use `Path.GetFileNameWithoutExtension`?

Reply:
Having looked at the docs "This is usually, but not necessarily, the file name..." so I'll change it to use the path.